cirq/testing/asynchronous.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2019 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from collections import Awaitable, Coroutine
+from typing import Any, Type, Union
+
+import pytest
+
+_just_return_result = ([],)  # type: Any
+
+
+def assert_asyncio_still_running(
+        future: Union[Awaitable, asyncio.Future, Coroutine],
+        timeout: float = 0.001):
+    """Checks that the given asyncio future has not completed.
+
+    Works by running the asyncio event loop for a short amount of time.
+
+    Args:
+        future: The future that should not yet be resolved.
+        timeout: The number of seconds to wait for the future. Make sure this is
+             a small value, because it holds up the passing test!
+
+    Raises:
+        AssertError: The future completed or failed within the timeout.
+    """
+    try:
+        asyncio.get_event_loop().run_until_complete(
+            asyncio.wait_for(asyncio.shield(future), timeout=0.001))
```

Comment:
nit: could factor out this snippet into a helper method, since it appears in all of these tests:
```python
def _run_until_complete(future, timeout):
    return asyncio.get_event_loop().run_until_complete(
        asyncio.wait_for(asyncio.shield(future), timeout=timeout)
```

Reply:
Done.